node/lib/util/cherry_pick_util.js

```diff
@@ -240,7 +273,20 @@ exports.computeChanges = co.wrap(function *(repo, index, targetCommit) {
         }
         return new ConflictEntry(entry.mode, entry.id.tostrS());
     }
+
+    const mergeBases = yield GitUtil.mergeBases(repo, head, targetCommit);
+    const submoduleShasInBases = {};
+    for (const base of mergeBases) {
+        const commit = yield NodeGit.Commit.lookup(repo, base);
+        const names = yield SubmoduleUtil.getSubmoduleNamesForCommit(repo,
+                                                                     commit);
+        const shas = yield SubmoduleUtil.getSubmoduleShasForCommit(repo, names,
```

Comment:
This is actually a fairly expensive operation; it has to call into libgit2 once for each submodule; it can be ~1s to do this for a meta-repo with thousands of submodules and I've gone to lengths in recent changes to avoid doing this for all subs.

Could you tweak this solution so that we only need to read the SHAs for subs in add-add situations? We should avoid calling `mergeBases` unless we're in this situation as well as we've seen that it can be pretty expensive (e.g., 800ms) even for branches just a week apart.